core/corehttp/gateway_handler.go

```diff
@@ -560,6 +565,11 @@ func (i *gatewayHandler) deleteHandler(w http.ResponseWriter, r *http.Request) {
 			return
 		}
 
+		if err := i.node.Providers.Provide(c); err != nil {
```

Comment:
@magik6k I have a question: why `c` here? It seems like this will try to provide the same `Cid` for each iteration of the loop. Is that intentional? If so, I'm curious why. I'm using this PR to inform a similar change I'm making and am caught up on this detail. Thanks.

Reply:
Yeah, this is wrong. This should be either `newnode.Cid()` or more likely not at all here (`Provide` below should handle this, not 100% sure)